backend/app/nodes/llm/generative/best_of_n.py

```diff
@@ -0,0 +1,234 @@
+from typing import Dict, List
+
+from pydantic import Field
+
+from ..single_llm_call import SingleLLMCallNodeConfig
+from ...subworkflow.ephemeral_subworkflow_node import EphemeralSubworkflowNode
+from ....schemas.workflow_schemas import (
+    WorkflowDefinitionSchema,
+    WorkflowNodeSchema,
+    WorkflowLinkSchema,
+)
+from ..llm_utils import LLMModels, ModelInfo
+
+
+class BestOfNNodeConfig(SingleLLMCallNodeConfig):
+    samples: int = Field(
+        default=3, ge=1, le=10, description="Number of samples to generate"
+    )
+    rating_prompt: str = Field(
+        default=(
+            "Rate the following response on a scale from 0 to 10, where 0 is poor "
+            "and 10 is excellent. Consider factors such as relevance, coherence, "
+            "and helpfulness. Respond with only a number."
+        ),
+        description="The prompt for the rating LLM",
+    )
+    rating_model_info: ModelInfo = Field(
+        default_factory=lambda: ModelInfo(
+            model=LLMModels.GPT_4O, max_tokens=16, temperature=0.1
+        ),
+        description="Model info for the rating LLM",
+    )
+    generation_llm_info: ModelInfo = Field(
+        default_factory=lambda: ModelInfo(
+            model=LLMModels.GPT_4O, max_tokens=150, temperature=0.7
+        ),
+        description="Model info for the generation LLM",
+    )
+    system_message: str = Field(
+        default="You are a helpful assistant.",
+        description="System message for the generation LLM",
+    )
+    user_message: str = Field(
+        default="{{ user_input }}", description="User message template"
+    )
+    input_schema: Dict[str, str] = Field(default={"user_input": "str"})
+    output_schema: Dict[str, str] = Field(default={"response": "str"})
+
+
+class BestOfNNode(EphemeralSubworkflowNode):
```

Comment:
could also consider using `Temporary` instead of `Ephemeral` for simplicity

Reply:
While, other workflows are **persisted** in the DB, these ones are created at runtime and not stored anywhere. 
I think "ephemeral" is a more apt term (and popular in software too) for such things, that are transient/not persistent.